lib/streamlit/runtime/secrets.py

```diff
@@ -164,6 +243,90 @@ def _reset(self) -> None:
+        """
+        secrets = {}
+
+        try:
+            with open(path, encoding="utf-8") as f:
+                secrets_file_str = f.read()
+        except FileNotFoundError:
+            return {}
+
+        try:
+            import toml
+
+            secrets.update(toml.loads(secrets_file_str))
+        except:
+            if print_exceptions:
+                st.error(
+                    secret_error_messages_singleton.get_message(
+                        SecretErrorMessagesKey.ERROR_PARSING_FILE_AT_PATH
+                    )(path)
+                )
+            raise
+
+        return secrets
+
+    def _parse_directory(self, path: str, print_exceptions: bool) -> Mapping[str, Any]:
+        """
+        Parse a directory for secrets. Directory style can be used to support Kubernetes secrets that are mounted to folders.
+
+        Example structure:
+        - top_level_secret_folder
+            - user_pass_secret (folder)
+                - username (file), content: myuser
+                - password (file), content: mypassword
+            - my_plain_secret (folder)
+                - regular_secret (file), content: mysecret
+
+        See: https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/#create-a-pod-that-has-access-to-the-secret-data-through-a-volume
+        And: https://docs.snowflake.com/en/developer-guide/snowpark-container-services/additional-considerations-services-jobs#passing-secrets-in-local-container-files
+        """
+        secrets = {}
+        for dirname in os.listdir(path):
+            sub_folder_path = os.path.join(path, dirname)
+            if not os.path.isdir(sub_folder_path):
+                error_msg = secret_error_messages_singleton.get_message(
+                    SecretErrorMessagesKey.SUBFOLDER_PATH_IS_NOT_A_FOLDER
+                )(sub_folder_path)
+                if print_exceptions:
+                    st.error(error_msg)
+                raise ValueError(error_msg)
+            sub_secrets = {}
+
+            for filename in os.listdir(sub_folder_path):
+                file_path = os.path.join(sub_folder_path, filename)
+
+                # ignore folders
+                if os.path.isdir(file_path):
+                    continue
+
+                with open(file_path) as f:
+                    sub_secrets[filename] = f.read().strip()
+
+            if len(sub_secrets) == 1:
+                # if there's just one file, collapse it so it's directly under `dirname`
+                secrets[dirname] = sub_secrets[list(sub_secrets.keys())[0]]
+            else:
+                secrets[dirname] = sub_secrets
+        return secrets
+
+    def _parse_file_path(self, path: str, print_exceptions: bool) -> Mapping[str, Any]:
+        if path.endswith(".toml"):
+            return self._parse_toml_file(path, print_exceptions)
+        elif os.path.isdir(path):
+            return self._parse_directory(path, print_exceptions)
+        else:
```

Comment:
someone can put a file that's not toml or directory in the config and that's how we will reach the else.

Reply:
I think what @raethlein is saying is you don't need this in the else clause, but just unindent the code since the first two branches return?

Not a big deal, but wanted to explain.

Reply:
ah. i see. updated